misc/dbt-materialize/tests/adapter/test_empty.py

```diff
@@ -0,0 +1,20 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the LICENSE file at the
+# root of this repository, or online at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dbt.tests.adapter.empty.test_empty import BaseTestEmpty
+
+
+class TestMaterializeEmpty(BaseTestEmpty):
```

Comment:
The `--empty` flag creates a dependency on the `quickstart` cluster (similar to what we worked around in #23712). The implementation of the flag uses `WHERE FALSE LIMIT 0`, which is a no-op in PostgreSQL, but requires a valid cluster in Materialize.

```bash
❯ dbt run --exclude config.materialized:source --empty
08:28:38  Running with dbt=1.8.0
08:28:38  [WARNING]: Deprecated functionality
The `tests` config has been renamed to `data_tests`. Please see
https://docs.getdbt.com/docs/build/data-tests#new-data_tests-syntax for more
information.
08:28:38  Registered adapter: materialize=1.8.0
08:28:38  Found 2 models, 1 test, 4 seeds, 1 source, 470 macros
08:28:38
NOTICE:  role default cluster "quickstart" does not exist
HINT:  Change the default cluster for the current role with `ALTER ROLE <role> SET cluster TO <cluster>;`.
NOTICE:  role default cluster "quickstart" does not exist
HINT:  Change the default cluster for the current role with `ALTER ROLE <role> SET cluster TO <cluster>;`.
NOTICE:  role default cluster "quickstart" does not exist
HINT:  Change the default cluster for the current role with `ALTER ROLE <role> SET cluster TO <cluster>;`.
08:28:42  Concurrency: 1 threads (target='dev')
08:28:42
08:28:42  1 of 1 START sql materialized_view model public.content_tags_model ............. [RUN]
NOTICE:  role default cluster "quickstart" does not exist
HINT:  Change the default cluster for the current role with `ALTER ROLE <role> SET cluster TO <cluster>;`.
08:28:43  1 of 1 ERROR creating sql materialized_view model public.content_tags_model .... [ERROR in 1.15s]
08:28:43
08:28:43  Finished running 1 materialized view model in 0 hours 0 minutes and 4.76 seconds (4.76s).
08:28:43
08:28:43  Completed with 1 error and 0 warnings:
08:28:43
08:28:43    Database Error in model content_tags_model (models/models/content_tags_model.sql)
  unknown cluster 'quickstart'
  compiled Code at target/run/mz_get_started/models/models/content_tags_model.sql
08:28:43
08:28:43  Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
```

We might need a similar workaround here.

Reply:
We can also address this as a follow-up.

Reply:
I was just looking into this. I think that the example that you have here is failing because you are trying to create a materialized view on an non-existing cluster rather than the `WHERE FALSE LIMIT 0`. Can you try adding `--log-level debug` to see the complete generated statement?

I've tested this with a non-materialized view and a non-existing default cluster and it works as expected:

```
=> show clusters;
       name       |   replicas   
------------------+--------------
 mz_introspection | r1 (3xsmall)
 mz_probe         | r1 (3xsmall)
 mz_support       | 
 mz_system        | r1 (3xsmall)

---

$ dbt run --select my_model --empty --log-level debug

...
18:32:35  Finished running 1 view model in 0 hours 0 minutes and 1.48 seconds (1.48s).
18:32:35  
18:32:35  Completed successfully

---

# Generated SQL:
create view "materialize"."public"."my_model"
  as (
select
    tested_column from (select * from "materialize"."public"."my_upstream_model" where false limit 0) _dbt_limit_subq_my_upstream_model
  );
```

Reply:
Or maybe this is just working as intended? I guess there is no way to materialize models that need to be associated with a cluster without requiring that a valid cluster exists, and I expected this command to allow overriding the materialization that is used to _e.g._ just materialize all models as views.

Reply:
> Or maybe this is just working as intended? I guess there is no way to materialize models that need to be associated with a cluster without requiring that a valid cluster exists, and I expected this command to allow overriding the materialization that is used to e.g. just materialize all models as views.

I think so too, if we were to extend the function and override it to convert all materialized views to just views, this would be a bit abstracted and hidden for the end users which could be quite misleading.